File: backend/src/main/java/com/ecommerce/backend/dto/view/OrderItemView.java
package com.ecommerce.backend.dto.view;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OrderItemView {

    private Long id;

    // tên sản phẩm / biến thể – tuỳ bạn muốn hiển thị thế nào
    private String productName;

    private String size;
    private String color;
    private String material;

    private String imageUrl;

    private double priceAtPurchase;
    private int quantity;
}

File: frontend_client_mobile/lib/models/order_item_view.dart
import 'package:json_annotation/json_annotation.dart';
part 'order_item_view.g.dart';

@JsonSerializable()
class OrderItemView {
  final int id;
  final String productName;
  final String? size;
  final String? color;
  final String? material;
  final String? imageUrl;
  final double priceAtPurchase;
  final int quantity;

  OrderItemView({
    required this.id,
    required this.productName,
    this.size,
    this.color,
    this.material,
    this.imageUrl,
    required this.priceAtPurchase,
    required this.quantity,
  });

  factory OrderItemView.fromJson(Map<String, dynamic> json) {
    return OrderItemView(
      id: json['id'],
      productName: json['productName'] ?? '',
      size: json['size'],
      color: json['color'],
      material: json['material'],
      imageUrl: json['imageUrl'],
      priceAtPurchase: (json['priceAtPurchase'] ?? 0).toDouble(),
      quantity: json['quantity'] ?? 0,
    );
  }
}